controllers/front/ProductController.php

```diff
@@ -576,6 +580,56 @@ protected function getMinimalProductOrDeclinationQuantity($product)
         return $minimalProductQuantity;
     }
 
+    /**
+     * Displays information, if the customer has this product in cart already.
+     */
+    protected function addCartQuantityNotification()
+    {
+        // Get quantity of this product in cart, it will return an array with
+        // quantity of this single product and also quantity in packs
+        $cartProductQuantity = $this->context->cart->getProductQuantity(
+            $this->id_product
+        );
+
+        // If there is nothing in the cart for this product, nothing to do here
+        if ($cartProductQuantity['deep_quantity'] == 0) {
+            return;
+        }
+
+        // Now, we got from this function both total count and standalone count, works like this:
+        // If you have 1 standalone - quantity 1, deep_quantity 1
+        // If you have 1 standalone and 1 in pack - quantity 1, deep_quantity 2
+        // If you have 1 in pack - quantity 0, deep_quantity 1
+        $standaloneQuantity = (int) $cartProductQuantity['quantity'];
+        $quantityInPacks = 0;
+
+        // If we have some products in pack only, we will subtract it to
+        if ($cartProductQuantity['quantity'] != $cartProductQuantity['deep_quantity']) {
+            $quantityInPacks = (int) $cartProductQuantity['deep_quantity'] - (int) $cartProductQuantity['quantity'];
+        }
+
+        // Render nice notifications so the user knows what is happening
+        if ($standaloneQuantity > 0 && $quantityInPacks > 0) {
+            $this->info[] = $this->trans(
+                'You have %1s piece(s) of this product in cart and another %2s piece(s) included in packs.',
+                [$standaloneQuantity, $quantityInPacks],
+                'Shop.Theme.Catalog'
+            );
+        } elseif ($standaloneQuantity > 0) {
+            $this->info[] = $this->trans(
+                'You have %1s piece(s) of this product in cart.',
```

Comment:
Piece feels weird item, quantity (hard to make it plural as (s)) or unit ?

Reply:
It's doable for v9 if you push this topic - https://github.com/PrestaShop/PrestaShop/pull/32940/files (https://github.com/PrestaShop/PrestaShop/issues/15870)